src/modules/aws_vpc/mappers/network_acl.ts

```diff
@@ -0,0 +1,205 @@
+              Key: k,
+              Value: e.tags![k],
+            };
+          });
+          input.TagSpecifications = [
+            {
+              ResourceType: 'network-acl',
+              Tags: tags,
+            },
+          ];
+        }
+        const res = await this.createNetworkAcl(client.ec2client, input);
+        if (res) {
+          // now we need to add the entries
+          for (const entry of e.entries ?? []) {
+            await this.createNetworkAclEntry(client.ec2client, entry);
+          }
+        }
+
+        // re-read the network acl to get the updated results
+        const rawAcl = await this.getNetworkAcl(client.ec2client, res?.NetworkAclId);
+        if (rawAcl) {
+          const newAcl = await this.networkAclMapper(rawAcl, e.region, ctx);
+          if (newAcl) {
+            newAcl.id = e.id;
+            await this.db.update(newAcl, ctx);
+            out.push(newAcl);
+          }
+        }
+      }
+      return out;
+    },
+    read: async (ctx: Context, id?: string) => {
+      if (!!id) {
+        const { networkAclId, region } = this.idFields(id);
+        const client = (await ctx.getAwsClient(region)) as AWS;
+        const rawAcl = await this.getNetworkAcl(client.ec2client, networkAclId);
+        if (!rawAcl || rawAcl.IsDefault) return undefined; // cannot read default acls
+        return await this.networkAclMapper(rawAcl, region, ctx);
+      } else {
+        const out: NetworkAcl[] = [];
+        const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
+        await Promise.all(
+          enabledRegions.map(async region => {
+            const client = (await ctx.getAwsClient(region)) as AWS;
+            for (const eg of await this.getNetworkAcls(client.ec2client)) {
+              if (eg && !eg.IsDefault) {
+                // skip default acls
+                const outEg = await this.networkAclMapper(eg, region, ctx);
+                if (outEg) out.push(outEg);
+              }
+            }
+          }),
+        );
+        return out;
+      }
+    },
+    updateOrReplace: (a: NetworkAcl, b: NetworkAcl) => {
+      // if we have modified vpc we need to replace
+      if (!Object.is(a.vpc?.vpcId, b.vpc?.vpcId)) return 'replace';
+      return 'update';
+    },
+    update: async (es: NetworkAcl[], ctx: Context) => {
+      const out: NetworkAcl[] = [];
+      for (const e of es) {
+        const client = (await ctx.getAwsClient(e.region)) as AWS;
+        const cloudRecord = ctx?.memo?.cloud?.NetworkAcl?.[this.entityId(e)];
+        const isUpdate = this.module.networkAcl.cloud.updateOrReplace(cloudRecord, e) === 'update';
+        if (isUpdate) {
+          let update = false;
+          if (!isEqual(cloudRecord.entries, e.entries)) {
+            // remove all non-default rules and recreate them
+            for (const cloudRule of cloudRecord.entries ?? []) {
+              if (!cloudRule.RuleNumber || cloudRule.RuleNumber > 32766) continue;
```

Comment:
nit: create a const for this magic number.

Reply:
also, does it make sense to add this as a constraint?

Reply:
will do

Reply:
the entries is set as a JSON type, mapped to AWS type directly of NetworkAclEntry. Cannot add the constraint in typeorm